src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,546 @@
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+
+        // Use the existing session for further processing.
+        outSession = *existingSession;
+    }
+
+    // Build and send the ProvideOfferResponse.
+    Commands::ProvideOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.videoStreamID.SetValue(outSession.videoStreamID);
+    resp.audioStreamID.SetValue(outSession.audioStreamID);
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+}
+
+void WebRTCTransportProviderServer::HandleProvideAnswer(HandlerContext & ctx, const Commands::ProvideAnswer::DecodableType & req)
+{
+    // Extract command fields from the request.
+    uint16_t sessionId = req.webRTCSessionID;
+    auto sdpSpan       = req.sdp;
+
+    NodeId peerNodeId = GetNodeIdFromCtx(ctx.mCommandHandler);
+
+    WebRTCSessionStruct * existingSession = FindSession(sessionId);
+
+    // Respond with NOT_FOUND if the session is not valid.
+    if (!existingSession)
+    {
+        ChipLogError(Zcl, "HandleProvideAnswer: No existing session with ID=%u.", sessionId);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::NotFound);
+        return;
+    }
+
+    // Verify that the session's PeerNodeID and FabricIndex match those from the secure session context.
+    if (peerNodeId != existingSession->peerNodeID)
+    {
+        ChipLogError(Zcl, "HandleProvideAnswer: Session ID=%u does not match PeerNodeID or FabricIndex.", sessionId);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::NotFound);
+        return;
+    }
+
+    std::string sdpAnswer(sdpSpan.data(), sdpSpan.size());
+    ctx.mCommandHandler.AddStatus(
+        ctx.mRequestPath, Protocols::InteractionModel::ClusterStatusCode(mDelegate.HandleProvideAnswer(sessionId, sdpAnswer)));
+}
+
+void WebRTCTransportProviderServer::HandleProvideICECandidates(HandlerContext & ctx,
+                                                               const Commands::ProvideICECandidates::DecodableType & req)
+{
+    // Extract command fields from the request.
+    uint16_t sessionId = req.webRTCSessionID;
+
+    NodeId peerNodeId = GetNodeIdFromCtx(ctx.mCommandHandler);
+
+    std::vector<std::string> candidates;
+    auto iter = req.ICECandidates.begin();
+    while (iter.Next())
+    {
+        // Get current candidate CharSpan and convert to std::string.
+        const CharSpan & candidateSpan = iter.GetValue();
+        candidates.emplace_back(candidateSpan.data(), candidateSpan.size());
+    }
+
+    // Check the validity of the list.
+    CHIP_ERROR listErr = iter.GetStatus();
+    if (listErr != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "HandleProvideICECandidates: ICECandidates list error: %s", chip::ErrorStr(listErr));
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
```

Comment:
The correct error is InvalidCommand, since the command is malformed, no?  Relevant IM spec text:

> If a mandatory data field is missing, or incoming data cannot be mapped to the expected data type for a field, a CommandStatusIB SHALL be generated with an error status of INVALID_COMMAND, even if the cluster defines another type of response.